In [2]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np
import requests
from bs4 import BeautifulSoup

import matplotlib.pyplot as plt
from matplotlib import font_manager, rc
from matplotlib import style
style.use('ggplot')
%matplotlib inline

font_name = font_manager.FontProperties(fname="c:/Windows/Fonts/malgun.ttf").get_name()
rc('font', family=font_name)
plt.rcParams['axes.unicode_minus'] = False


from IPython.display import set_matplotlib_formats
set_matplotlib_formats("retina")

In [68]:
cx, cy = map(str, input().split())

saga_name = []
juso = []
upjong_code = []
hjdong = []

open_api_key = 'oh3n93%2BmUwsTMmVjqBVN%2BJPSdFnw5%2F%2FET46RfYbH%2F4uLXLWCkAXRdI16yhhovH%2FYUP9EiHMtDoW%2FVeGzZTDvpg%3D%3D'
params = '&radius=500&cx='+cx+'&cy='+cy+'&indsMclsCd=D03&numOfRows=100&'
open_url = 'http://apis.data.go.kr/B553077/api/open/sdsc/storeListInRadius?ServiceKey=' + open_api_key + params

res = requests.get(open_url)
soup = BeautifulSoup(res.content, 'html.parser')

data = soup.find_all('item')

for item in data :
    saga_name.append(item.select_one("bizesnm").get_text())
    juso.append(item.select_one("rdnm").get_text())
    hjdong.append(item.select_one("adongnm").get_text())
    upjong_code.append(item.select_one("indsSclsCd").get_text())
    
df = pd.DataFrame({'상가이름': saga_name,'업종코드':upjong_code, '행정동':hjdong, '주소': juso})

#종합소매, 편의점, 할인점, 수퍼마켓 아닌 업종 추출
for item in df['업종코드']:
    del_list = []
    if (item!='D03A06') and (item != 'D03A01') and (item!= 'D03B10') and (item != 'D03A02'):
        del_list.append(item)

129.084548 35.238838 


* 예시 위경도 129.084548 35.238838 

* D	소매	D03	종합소매점	D03A06	종합소매
* D	소매	D03	종합소매점	D03A01	편의점
* D	소매	D03	종합소매점	D03B10	할인점
* D	소매	D03	종합소매점	D03A02	수퍼마켓